pkg/ui/src/views/statements/statementsPage.tsx

```diff
@@ -0,0 +1,215 @@
+import { ColumnDescriptor, SortedTable } from "src/views/shared/components/sortedtable";
+import { SortSetting } from "src/views/shared/components/sortabletable";
+import { refreshQueries } from "src/redux/apiReducers";
+import { QueriesResponseMessage } from "src/util/api";
+import { appAttr } from "src/util/constants";
+import { Duration } from "src/util/format";
+import { summarize, StatementSummary } from "src/util/sql/summarize";
+
+import { countBarChart, rowsBarChart, latencyBarChart } from "./barCharts";
+
+import * as protos from "src/js/protos";
+import "./statements.styl";
+
+type CollectedStatementStatistics$Properties = protos.cockroach.sql.CollectedStatementStatistics$Properties;
+
+class StatementsSortedTable extends SortedTable<CollectedStatementStatistics$Properties> {}
+
+interface StatementsPageProps {
+  statements: CachedDataReducerState<QueriesResponseMessage>;
+  refreshQueries: typeof refreshQueries;
+}
+
+interface StatementsPageState {
+  sortSetting: SortSetting;
+}
+
+function StatementLink(props: { statement: string }) {
+  const summary = summarize(props.statement);
+
+  return (
+    <Link to={ `/statement/${encodeURIComponent(props.statement)}` }>
+      <div title={ props.statement }>{ shortStatement(summary, props.statement) }</div>
+    </Link>
+  );
+}
+
+function AppLink(props: { app: string }) {
+  return (
+    <Link to={ `/statements/${encodeURIComponent(props.app)}` }>
+      { props.app }
+    </Link>
+  );
+}
+
+function shortStatement(summary: StatementSummary, original: string) {
+  switch (summary.statement) {
+    case "update": return "UPDATE " + summary.table;
+    case "insert": return "INSERT INTO " + summary.table;
+    case "select": return "SELECT FROM " + summary.table;
+    case "delete": return "DELETE FROM " + summary.table;
+    case "create": return "CREATE TABLE " + summary.table;
+    default: return original;
+  }
+}
+
+function calculateCumulativeTime(query: CollectedStatementStatistics$Properties) {
+  const count = FixLong(query.stats.count).toInt();
+  const latency = query.stats.service_lat.mean;
+
+  console.log(latency, count, count * latency);
+
+  return count * latency;
+}
+
+function makeStatementsColumns(statements: CollectedStatementStatistics$Properties[])
+    : ColumnDescriptor<CollectedStatementStatistics$Properties>[] {
+  const countBar = countBarChart(statements);
+  const rowsBar = rowsBarChart(statements);
+  const latencyBar = latencyBarChart(statements);
+
+  return [
+    {
+      title: "App",
+      cell: (query) => <AppLink app={ query.key.app } />,
+      sort: (query) => query.key.app,
+    },
+    {
+      title: "Statement",
+      className: "statements-table__col-query-text",
+      cell: (query) => <StatementLink statement={ query.key.query } />,
+      sort: (query) => query.key.query,
+    },
+    {
+      title: "Time",
+      cell: (query) => Duration(calculateCumulativeTime(query) * 1e9),
```

Comment:
Love this column, but seems weird for it to not have a bar chart like the others. I know we're running out of horizontal space; may have to do something clever down the road to regain space like putting stat numbers on top of bars instead of next to them, etc.